python-client/tira/evaluators.py

```diff
@@ -0,0 +1,258 @@
+            ret["RR"] = te.get_reciprocal_rank()
+
+        return {k: ret[k] for k in self.measures}
+
+
+EVALUATORS = {
+    "TrecTools": TrecToolsEvaluator,
+    "RunFileEvaluator": RunFileEvaluator,
+    "HuggingFaceEvaluator": HuggingFaceEvaluator,
+}
+
+MEASURE_TO_EVALUATORS = {
+    "nDCG@10": "TrecTools",
+    "RR": "TrecTools",
+    "P@10": "TrecTools",
+    "Docs Per Query (Avg)": "RunFileEvaluator",
+    "Docs Per Query (Min)": "RunFileEvaluator",
+    "Docs Per Query (Max)": "RunFileEvaluator",
+    "NumQueries": "RunFileEvaluator",
+    "accuracy": "HuggingFaceEvaluator",
+    "recall": "HuggingFaceEvaluator",
+    "precision": "HuggingFaceEvaluator",
+    "f1": "HuggingFaceEvaluator",
+}
+
+
+def load_evaluator_config(config: Union[dict, str], client: Optional[TiraClient] = None):
+    if isinstance(config, str):
+        if client is None:
+            client = Client()
+        dataset_config = client.get_dataset(config)
+        if "trusted_evaluator" not in dataset_config or not dataset_config["trusted_evaluator"]:
+            raise ValueError(f'No trusted evaluation is configured for the dataset "{config}".')
+
+        return load_evaluator_config(dataset_config["trusted_evaluator"])
+
+    if "measures" not in config or not config["measures"]:
+        raise ValueError("Configuration of the evaluator is invalid: No measures are specified.")
+
+    if "run_format" not in config:
+        raise ValueError("Configuration of the evaluator is invalid: No run_format is specified.")
+
+    if "truth_format" not in config:
+        raise ValueError("Configuration of the evaluator is invalid: No truth_format is specified.")
+
+    return config
+
+
+def get_evaluators_if_valid(config: Union[dict, str], client: Optional[TiraClient] = None):
+    config = load_evaluator_config(config, client)
+
+    evaluator_to_measures = {}
+    for measure in config["measures"]:
+        evaluator = MEASURE_TO_EVALUATORS[measure]
+        if evaluator not in evaluator_to_measures:
+            evaluator_to_measures[evaluator] = []
+        evaluator_to_measures[evaluator] += [measure]
+
+    ret = []
+    for evaluator, measures in evaluator_to_measures.items():
+        impl = EVALUATORS[evaluator](config["run_format"], config["truth_format"], measures)
+        impl.configuration_is_valid(config["run_format"], config["truth_format"], config)
+        ret.append(impl)
+
+    return ret
+
+
+def evaluate(
+    run: Path,
+    truths: Path,
+    config: Union[dict, str],
+    output_dir: Optional[Path] = None,
+    client: Optional[TiraClient] = None,
+):
+    config = load_evaluator_config(config, client)
+    evaluators = get_evaluators_if_valid(config, client)
+    ret = {}
+    for evaluator in evaluators:
+        evaluation = evaluator.evaluate(run, truths)
+        ret.update(evaluation)
+
+    if output_dir:
+        prototext = to_prototext([ret])
+        with open(output_dir / "evaluation.prototext") as f:
+            f.write(prototext)
```

Comment:
Since prototext is already a string (I asssume), we don't need to open the stream ourselves and can instead use:
```suggestion
        (output_dir / "evaluation.prototext").write_text(prototext)
```